components/button/theme.css

```diff
@@ -1,208 +1,406 @@
+  --button-border-radius: calc(.4 * var(--unit));
+  --button-margin: calc(0.6 * var(--unit));
   --button-icon-margin: calc(0.6 * var(--unit));
-  --button-squared-min-width: calc(4 * var(--unit));
-}
-
-
-@keyframes spin {
-  0% {
-    transform: rotate(0deg);
-  }
-  100% {
-    transform: rotate(359deg);
-  }
+  --button-icon-children-margin: calc(0.6 * var(--unit));
+  --spinner-border-width: 2px;
 }
 
 .button {
-  @apply --reset;
-  @apply --no-text-decoration;
-  align-content: center;
   align-items: center;
-  border: 0;
+  background: var(--background-color);
+  border: var(--border);
   border-radius: var(--button-border-radius);
+  box-shadow: var(--box-shadow);
+  color: var(--text-color);
   cursor: pointer;
   display: inline-block;
-  flex-direction: row;
-  font-size: calc(1.4 * var(--unit));
-  font-weight: var(--button-font-weight);
-  height: var(--button-small-height);
-  line-height: var(--button-small-line-height);
-  justify-content: center;
+  display: inline-flex;
+  font-size: var(--font-size);
+  font-family: var(--font-family-semi-bold);
+  height: var(--height);
+  line-height: var(--line-height);
   letter-spacing: 0;
+  min-width: var(--height);
   outline: none;
-  padding: 0;
-  position: relative;
+  padding: var(--padding);
   text-align: center;
   text-decoration: none;
-  text-transform: uppercase;
-  transition: background-color 0.2s var(--animation-curve-default),
-  color .2s var(--animation-curve-default);
+  transition: background-color 0.2s cubic-bezier(0.4, 0, 0.2, 1),
+    color .2s cubic-bezier(0.4, 0, 0.2, 1);
+  user-select: none;
   white-space: nowrap;
+  vertical-align: middle;
 
-  &:not(.iconOnly) .icon,
-  &:not(.iconOnly) > svg {
-    margin-right: var(--button-icon-margin);
-  }
-
-  & .icon {
-    font-size: 120%;
-    vertical-align: middle;
+  .children {
+    padding-top: var(--button-children-padding-correction);
   }
 
-  & > input {
-    height: 0.1px;
-    margin: 0;
-    opacity: 0;
-    overflow: hidden;
-    padding: 0;
-    position: absolute;
-    width: 0.1px;
-    z-index: 0;
+  * + .children,
+  .children + *:not(.spinner) {
+    margin-left: var(--button-icon-children-margin);
   }
 
   &::-moz-focus-inner {
     border: 0;
   }
 
-  & > span:not([data-teamleader-ui='tooltip']) {
-    display: inline-block;
-    line-height: calc(3 * var(--unit));
-    vertical-align: top;
-  }
-
-  & > svg {
+  svg {
+    color: inherit;
     display: inline-block;
     fill: currentColor;
-    font-size: 120%;
-    height: calc(3 * var(--unit));
-    vertical-align: top;
-    width: 1em;
+
+    * {
+      fill: currentColor;
+    }
   }
 
   & > * {
     pointer-events: none;
   }
 
   &[disabled] {
-    color: var(--button-disabled-text-color);
     cursor: auto;
     pointer-events: none;
   }
 }
 
-.bordered {
-  border: 1px solid var(--button-border-color);
+/* Button colors */
+.iconButton,
+.neutral,
+.mint,
+.outline,
+.ruby {
```

Comment:
Q: do you need these classes? Because these classes just resemble all the possible button types? So they seem obsolete to me, and I think you can just put the content under `.button`.

Reply:
You are right, I could just put the content under .button!